### ISSUE TITLE
duplicate vkCmdBind calls on record

### ISSUE THREAD

**Issue by theodoregoetz** [vsg-dev/VulkanSceneGraph#1170]
**Describe the bug**
vkCmdBindPipeline and vkCmdBindDescriptorSets (and potentially others) were recorded twice - first with the State stack and then again with the StateCommand which was processed as a Command object.

**To Reproduce**
Using renderdoc on any VSG example shows two vkCmdBindPipeline calls were recorded whenever the state group changes.

**Expected behavior**
Only a single vkCmdBind* call is needed per pipeline or descriptor set.

**Screenshots**
example using renderdoc showing duplicate bind calls recorded:
![vsg-bind-bug](https://github.com/vsg-dev/VulkanSceneGraph/assets/3765682/05b3adee-1815-4d4d-8202-b00bd7b5e6c6)

example showing expected behavior:
![vsg-bind-expected](https://github.com/vsg-dev/VulkanSceneGraph/assets/3765682/cb0e611f-3d30-4d53-a433-64bc30b36dc6)

**Desktop:**
 - Fedora 39
 - Nvidia GPU

**Comment by robertosfield** [comment]
Just so happens that I saw some odd results yesterday when looking at the API output but didn't spot why it was happening, and this Issue report actually pinpoints it.

I have had a quick look at the PR you've made and while this will fix it for your usage case it'll break valid usage.  I suspect a change I made elsewhere in the VSG has caused this regression so I'll investigate this morning and hopefully get a fix checked in.
